### PR TITLE
Add validation support for React Aria Components

### PR DESCRIPTION
**Pull request by devongovett**
Depends on #5288

This adds support for the validation behaviors described in #5288 to React Aria Components. It includes two new components:

* `<FieldError>` renders form validation errors. It accepts render props to allow customization of the error messages.
* `<Form>` is a small wrapper around an HTML `<form>` that also supports the `validationErrors` prop for server errors, which is provided via the `FormValidationContext`. We could also potentially add more functionality here in the future (e.g. auto focusing the first invalid field).

In addition, in RAC `validationBehavior="native"` is the default rather than `validationBehavior="aria"`. I think this is a better default, but we cannot change the behavior in RSP for backward compatibility reasons.

### CODE REVIEW

**Comment by LFDanLu** [comment]
mind rebasing this, looks like it has all the changes from https://github.com/adobe/react-spectrum/pull/5288 as well

File: packages/react-aria-components/src/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+export interface FormProps extends FormHTMLAttributes<HTMLFormElement> {
+  /**
+   * Validation errors for the form, typically returned by a server.
+   * This should be set to an object mapping from input names to errors.
+   */
+  validationErrors?: ValidationErrors | null
+}
+
+function Form(props: FormProps, ref: ForwardedRef<HTMLFormElement>) {
```
Comment (snowystinger) [inline packages/react-aria-components/src/Form.tsx:25]: Description says small wrapper and can add in the future, so I assume suggestions like providing data attributes and renderprops for focusWithin or a FormContext for passing in props would fall into that category of enhancements?

**Review by snowystinger — DISMISSED** [review]

File: packages/react-aria-components/src/Form.tsx (hunk 1 above)
Comment (devongovett) [reply packages/react-aria-components/src/Form.tsx:25]: good questions. I'm not sure what the use case for focus within on a form would be. I thought our recommendation was to use Group in those cases. I guess we could support FormContext for consistency but not really sure what people would use it for.

Comment (snowystinger) [reply packages/react-aria-components/src/Form.tsx:25]: Yeah, I realized you could just use a Group, though don't know if it'd be a nice convenience feature

Comment (snowystinger) [reply packages/react-aria-components/src/Form.tsx:25]: We do very explicitly say
> All React Aria Components export a corresponding context that can be used to send props to them from a parent element

in our docs, just noticed

**Review by LFDanLu — APPROVED** [review]

**Comment by LFDanLu** [comment]
ah, didn't see the 2nd merge, looks like the tests need to be updated to have the new Item changes from the other PR. Will reapprove after that is updated

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**